dask_cuda/local_cuda_cluster.py

```diff
@@ -129,16 +129,29 @@ def new_worker_spec(self):
             )
 
         spec = copy.deepcopy(self.new_spec)
-        ii = self.cuda_visible_devices.index(name)
+        worker_count = self.cuda_visible_devices.index(name)
+        visible_devices = cuda_visible_devices(worker_count, self.cuda_visible_devices)
         spec["options"].update(
             {
-                "env": {
-                    "CUDA_VISIBLE_DEVICES": cuda_visible_devices(
-                        ii, self.cuda_visible_devices
-                    )
-                },
-                "plugins": {CPUAffinity(get_cpu_affinity(ii))},
+                "env": {"CUDA_VISIBLE_DEVICES": visible_devices,},
+                "plugins": {CPUAffinity(get_cpu_affinity(worker_count))},
             }
         )
-
+        # If TopologicalDistance from ucp is available, we set the UCX
+        # net device to the closest network device explicitly.
+        try:
+            from ucp._libs.topological_distance import TopologicalDistance
+
+            dev = int(visible_devices.split(",")[0])
+            net_dev = ""
+            td = TopologicalDistance()
+            ibs = td.get_cuda_distances_from_device_index(dev, "openfabrics")
+            if len(ibs) > 0:
+                net_dev += ibs[0]["name"] + ":1,"
+            ifnames = td.get_cuda_distances_from_device_index(dev, "network")
+            if len(ifnames) > 0:
+                net_dev += ifnames[0]["name"]
```

Comment:
Does it make sense to have multiple interfaces in `UCX_NET_DEVICES`, here we would have `"mlx5_0:1,enp1s0f0"`, do you know how UCX makes use of multiple interfaces? I haven't tried that so far to see what's the behavior.

Reply:
On DGX I actually gets `"mlx5_0:1,ib0"`. 
The ideas is that if UCX isn't configured with IB, it will use the TCP interface.

Reply:
You're right, `ib0` is the closest network device. Is `ib0` also valid for UCX? I've always seen the usage of `mlx5_*`, are both the same for UCX?